File: shared/main/java/net/ccbluex/liquidbounce/utils/timer/MSTimer.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.timer;

public final class MSTimer
{

	private long time = -1L;

	public boolean hasTimePassed(final long millis)
	{
		return System.currentTimeMillis() >= time + millis;
	}

	public long hasTimeLeft(final long millis)
	{
		return millis + time - System.currentTimeMillis();
	}

	public long getTime()
	{
		return System.currentTimeMillis() - time;
	}

	public void reset()
	{
		time = System.currentTimeMillis();
	}
}

File: src/main/java/net/ccbluex/liquidbounce/utils/timer/MSTimer.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.timer

class MSTimer
{
	private var time = -1L

	fun hasTimePassed(millis: Long): Boolean = System.currentTimeMillis() >= time + millis

	fun hasTimeLeft(millis: Long): Long = millis + time - System.currentTimeMillis()

	fun getTime(): Long = System.currentTimeMillis() - time

	fun reset()
	{
		time = System.currentTimeMillis()
	}
}
